### ISSUE TITLE
Project creation multiple issues

### ISSUE THREAD

**Issue by LB-BartoszDudzinski** [o3de/o3de#6955]
**Describe the bug**
Projects after creation are not added to the Project Manager and cannot be added manually afterwards. Configuring Gems via PM is also impossible as they are not available in the Configure Gems section. The only workaround is to create a project via CMD, however it only works when the "source engine" build model is used. Creating the Visual Studio project when the "Pre-built SDK engine" build model is used, stops without any errors after detecting CXX compile features.

The output:
```
C:\O3DE-install\NewPro>cmake -B build/windows2019 -S . -G "Visual Studio 16" -DLY_3RDPARTY_PATH=C:\o3de\o3de-packages -DLY_VERSION_ENGINE_NAME=o3de-install -DCMAKE_INSTAL_PREFIX=C:\O3DE-install
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19042.
-- The C compiler identification is MSVC 19.29.30040.0
-- The CXX compiler identification is MSVC 19.29.30040.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.29.30037/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.29.30037/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
```
The next line should be "-- Found o3de: TRUE", but it is not displayed.
This issue occurs on all platforms.

**Steps to reproduce**
1. Build the Engine.
2. Launch the Project Manager.
3. Create a project.
4. (optional) Navigate to the Configure Gems section during the creation process.

**Expected behavior**
The new project is created and added to the Project Manager.
The Gems list is available in the Configure Gems section.

**Actual behavior**
Projects are not added to the Project Manager. Gems are not listed in the Configure Gems section.

**Screenshots/Video**

https://user-images.githubusercontent.com/86952009/149767385-09a67e60-68cd-4398-b509-019b8ea636c2.mp4

**Found in Branch**
Development (https://github.com/o3de/o3de/commit/76ff7aec29ad67f11f0dc84e47702cc8757dcdd2)

**Desktop/Device:**
- Device: PC
- OS: Windows
- Version 10
- CPU Intel Core i5-9600k
- GPU Nvidia GeForce RTX 2060
- Memory 16GB

**Comment by lumberyard-employee-dm** [comment]
The fix for this issue was merged in PR #7001

**Comment by LB-BartoszDudzinski** [comment]
Verified as fixed in development
